### PR TITLE
Send transports as defined in the U2F JavaScript API.

### LINKED ISSUE #75: Server sends invalid values for transports field
When the server sends the transports field to the client, using the U2F JavaScript API version 1.1, it does so using values not defined in the API. Example request:

{"type":"u2f_register_request","appId":"https://u2fdemo.appspot.com","registerRequests":[{"challenge":"...","version":"U2F_V2"}],"registeredKeys":[{"appId":"https://u2fdemo.appspot.com","version":"U2F_V2","keyHandle":"...","transports":["BLUETOOTH_LOW_ENERGY"]}],"timeoutSeconds":30,"requestId":1}

This value, "BLUETOOTH_LOW_ENERGY", is not defined in the spec. (The defined value for Bluetooth Low Energy is "ble".)

### PR DESCRIPTION
**Pull request by juanlang**
Make the transports enum contain the string value used by the API.
Send the transports value as a comma-delimited string, rather than
a JSON array.

Closes #75.

### CODE REVIEW

File: u2f-ref-code/java/src/com/google/u2f/server/data/SecurityKeyData.java [hunk 1]
```diff
@@ -16,10 +16,21 @@
 
 public class SecurityKeyData {
   public enum Transports {
-    BLUETOOTH_RADIO,
-    BLUETOOTH_LOW_ENERGY,
-    USB,
-    NFC
+    BLUETOOTH_RADIO("bt"),
```
Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/data/SecurityKeyData.java]: Hmm.  ```BLUETOOTH_RADIO``` seems like a poor choice here.  We should probably call it ```BLUETOOTH_BREDR```

File: u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java [hunk 2]
```diff
@@ -100,15 +102,16 @@ public boolean equals(Object obj) {
         && Objects.equals(sessionId, other.sessionId);
   }
 
-  private JsonArray getTransportsAsJson() {
+  private String getTransportsAsString() {
```
Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java]: probably should put public method before private method.

File: u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java [hunk 3]
```diff
@@ -117,9 +120,9 @@ public JsonObject getJson() {
     result.addProperty("version", version);
     result.addProperty("keyHandle", keyHandle);
     result.addProperty("sessionId", sessionId);
-    JsonArray transportsJson = getTransportsAsJson();
-    if (transportsJson != null) {
-      result.add("transports", transportsJson);
+    String transportsString = getTransportsAsString();
+    if (transportsString != null) {
```
Comment (leshi) [inline u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java]: You need to test for empty string here I think.

```String.join(",", <empty list>)``` will return an empty string -- just verified.

Comment (juanlang) [inline u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java]: That's actually correct: if the transports are present (non-null) but empty, we want to send the empty string.

On a different note, the current source depends on Java7, but String.join only appears in Java8. I'll rewrite this line the uglier way.

File: u2f-ref-code/java/src/com/google/u2f/server/data/SecurityKeyData.java (hunk 1 above)
Comment (juanlang) [inline u2f-ref-code/java/src/com/google/u2f/server/data/SecurityKeyData.java]: Done.

File: u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java (hunk 2 above)
Comment (juanlang) [inline u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java]: Done.

**Comment by leshi** [comment]
lgtm
